beater/server.go

```diff
@@ -29,12 +19,12 @@ var (
 
 type reporter func([]beat.Event) error
 
-var (
-	errInvalidToken    = errors.New("Invalid token")
-	errPOSTRequestOnly = errors.New("Only post requests are supported")
-)
+type server struct {
+	http.Server
+	config Config
+}
```

Comment:
TBH this feels a bit like back & forth, because it is reverting some of the changes in PR https://github.com/elastic/apm-server/pull/78. I don't see the additional value of reverting this.
I generally still like the idea of grouping together methods, but as @jalvz pointed out earlier it does not make too much sense in this special case, since the `http.Server` offers the methods and it would basically just be a wrapper. Storing the config in the server stores quite some values as duplicates and adds more state to the server object. 
Could you quickly motivate this change a bit more so I can understand the value that it adds?

Reply:
The motivation is to have all the server logic "hidden" inside server. Config is only passed once on setup and from the on it's up to the server to decide when and where the config is needed.

Second part is the error handling. There was knowledge outside the server about how to handle specific errors which I don't think beater should know about. 

In general I think it's important that we can change the server without having to change anything in the beater `Run` method. 

The server object is composed of the http.Server. In my opinion this adds value as it simplifies the interface to interact with the server. 


For the duplicated config values: As the server instance only exists one I don't think this should be an issue.